src/vdaf/prio2.rs

```diff
@@ -87,6 +85,27 @@ impl Prio2 {
             Prio2PrepareShare(verifier_share),
         ))
     }
+
+    /// Choose a random point for polynomial evaluation.
+    ///
+    /// The point returned is not one of the roots used for polynomial interpolation.
+    pub(crate) fn choose_eval_at<S>(&self, prng: &mut Prng<FieldPrio2, S>) -> FieldPrio2
+    where
+        S: SeedStream,
+    {
+        // Make sure the query randomness isn't a root of unity. Evaluating the proof at any of
+        // these points would be a privacy violation, since these points were used by the prover to
+        // construct the wire polynomials.
+        let n = (self.input_len + 1).next_power_of_two();
+        let proof_length = 2 * n;
+        loop {
+            let eval_at: FieldPrio2 = prng.get();
+            // Unwrap safety: the constructor checks that this conversion succeeds.
```

Comment:
Since we aren't constrained by the old API anymore we might consider propagating this error instead of unwrapping.